presentation/src/main/java/com/example/presentation/view/CountriesScreen.kt

```diff
@@ -0,0 +1,100 @@
+package com.example.presentation.view
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import coil.decode.SvgDecoder
+import coil.request.ImageRequest
+import com.example.common.models.DataState
+import com.example.presentation.model.CountryUi
+
+@Composable
+fun CountriesScreen(
+    uiState: DataState<List<CountryUi>>,
+    onItemClick: (CountryUi) -> Unit,
```

Comment:
You should always try to pass only the necessary parameters. Here you need id from CountryUi so don't pass the entire instance